kingpin/actors/rightscale/server_array.py

```diff
@@ -996,6 +1002,19 @@ def _wait_until_healthy(self, array, sleep=60):
             if min_count <= count:
                 raise gen.Return()
 
+            all_instances = yield self._client.get_server_array_current_instances(array)
+            stranded_instances = yield self._client.get_server_array_current_instances(
+                array, filters=['state==stranded'])
+
+            self.log.info('Found %d stranded instances.' % len(stranded_instances))
```

Comment:
99% of the time there will be 0 stranded instances, but this line doubles the log lines. I suggest putting it into an if statement and then changing this to a warning instead of info